crates/graphql/src/schema/query_root.rs

```diff
@@ -101,11 +143,31 @@ impl QueryRoot {
         wallet: PublicKey<Wallet>,
         limit: i32,
         offset: i32,
+        exclude_types: Option<Vec<String>>,
     ) -> FieldResult<Vec<FeedEvent>> {
         let conn = ctx.shared.db.get().context("failed to connect to db")?;
 
-        let feed_events =
-            queries::feed_event::list(&conn, wallet, limit.try_into()?, offset.try_into()?)?;
+        let exclusion_types_parsed: Option<Vec<EventType>>;
+        if let Some(types) = exclude_types {
+            exclusion_types_parsed = Some(
+                types
+                    .iter()
+                    .map(|v| EventType::from_str(v))
+                    .filter(|v| !v.is_err())
+                    .map(Result::unwrap)
+                    .collect(),
+            );
+        } else {
+            exclusion_types_parsed = None;
+        }
```

Comment:
You can use .map on excluded_types which will return none if its none or transform the value. If you want to rewrap in an Option then and_then will be what you want.